File: src/main/java/monnef/core/power/WrenchHelper.java
/*
 * Jaffas and more!
 * author: monnef
 */

package monnef.core.power;

import monnef.core.power.IPipeWrench;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;

import java.util.HashSet;

public class WrenchHelper {
    private static HashSet<Integer> wrenchIDs = new HashSet<Integer>();

    public static void registerWrenchItem(int itemId){
        wrenchIDs.add(itemId);
    }

    public static boolean isPipeWrenchOrCompatible(Item item) {
        if (item instanceof IPipeWrench) return true;
        if(wrenchIDs.contains(item.itemID)) return true;
        return false;
    }

    public static boolean isHoldingWrench(EntityPlayer player) {
        ItemStack hand = player.getHeldItem();
        if (hand != null && isPipeWrenchOrCompatible(hand.getItem())) {
            return true;
        }
        return false;
    }
}

File: src/main/java/monnef/core/power/WrenchHelper.scala
/*
 * Jaffas and more!
 * author: monnef
 */
package monnef.core.power

import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import scala.collection.immutable.HashSet

object WrenchHelper {
  def registerWrenchItem(item: Item) {
    wrenches += item
  }

  def isPipeWrenchOrCompatible(item: Item): Boolean =
    if (item.isInstanceOf[IPipeWrench]) true
    else if (wrenches.contains(item)) true
    else false

  def isHoldingWrench(player: EntityPlayer): Boolean = {
    val hand: ItemStack = player.getHeldItem
    hand != null && isPipeWrenchOrCompatible(hand.getItem)
  }

  private var wrenches: HashSet[Item] = new HashSet[Item]
}